wright-filter-wheels/wright_filter_wheels/_yaqd_wright_filter_wheels_discrete.py

```diff
@@ -0,0 +1,101 @@
+        self._motornum=config["motor"]
+        self._serial_port = aserial.ASerial(config["serial_port"], config["baud_rate"])
+        self.microstep=1
+        self._steps_per_rotation=400
+        self._current_identifier=""
+        self._offset=0  # try incorporating any adjustments to TOML before adjusting this
+        self._identifiers=config["identifiers"]
+        self.home()
+
+
+    def _load_state(self, state):
+        """Load an initial state from a dictionary (typically read from the state.toml file).
+
+        Must be tolerant of missing fields, including entirely empty initial states.
+
+        Parameters
+        ----------
+        state: dict
+            The saved state to load.
+        """
+        super()._load_state(state)
+        # This is an example to show the symetry between load and get
+        # If no persistent state is needed, these unctions can be deleted
+        self.value = state.get("value", 0)
+
+    def get_state(self):
+        state = super().get_state()
+        state["value"] = self.value
+        return state
+
+    def set_offset(self,value):
+        self._offset=value
+
+    def set_identifier(self,key):
+        position_temp=self._identifiers[key]
+        self.set_position(position_temp)
+        self._approximate_position=position_temp
+        self._current_identifier=key
+
+    def get_position_identifiers(self):
+        return self._identifiers 
+
+    def get_identifier(self):
+        return self._current_identifier
+
+    def _set_position(self, position):
+        step_position=round(self.microstep*(position-self._position+self._offset)*self._steps_per_rotation/360)
+        self._serial_port.write(f"M {self._motornum} {step_position}\n".encode())
+        self._position=position
+
+    def direct_serial_write(self, message):
+        self._busy = True
+        self._serial_port.write(message.encode())
+
+    def home(self):
+        loop = asyncio.get_event_loop()
+        loop.create_task(self._home())
+
+    async def _home(self):
+        self._busy = True
+        # Initiate the home
+        self._serial_port.write(f"H {self._motornum}\n".encode())
+        await self._not_busy_sig.wait()
+        self._position=0
+        self._approximate_position=0
+        self._current_identifier=""
+
+    
+    def set_microstep(self, microint):
+        self._busy = True
+        if microint in [2**i for i in range(0,6)]:
+            self._serial_port.write(f"U {microint}\n".encode())
+            self.microstep=microint
+        
+
+    async def update_state(self):
+        while True:
+            # Perform any updates to internal state
+            self._serial_port.write(f"Q {self._motornum}\n".encode())
+            line = await self._serial_port.areadline()
+            self._busy = (line[0:1] != b"R")
+            #self.logger.debug(self._identifiers)
+            #await asyncio.sleep(2)
+            if self._busy:
+                await asyncio.sleep(0.1)
```

Comment:
```suggestion
                await asyncio.sleep(0.1)
            else:
                await self._busy_sig.wait()
```

Reply:
this may not yet be resolved and will be brought in next rev.   For now it is just considered a timing issue.